examples/imagenet/train_resnet_imagenet1k.py

```diff
@@ -0,0 +1,297 @@
+                    choices=['cross_entropy', 'binary_cross_entropy'])
+
+# Optimizer arguments
+parser.add_argument('--learning_rate', help='Optimizer learning rate', type=float, default=2.048)
+parser.add_argument('--momentum', help='Optimizer momentum', type=float, default=0.875)
+parser.add_argument('--weight_decay', help='Optimizer weight decay', type=float, default=5.0e-4)
+
+# LR scheduler arguments
+parser.add_argument('--t_warmup',
+                    help='Duration of learning rate warmup specified as a Time string',
+                    type=Time.from_timestring,
+                    default='8ep')
+parser.add_argument('--t_max',
+                    help='Duration to cosine decay the learning rate specified as a Time string',
+                    type=Time.from_timestring,
+                    default='1dur')
+
+# Save checkpoint arguments
+parser.add_argument('--save_checkpoint_dir',
+                    help='Directory in which to save model checkpoints',
+                    type=str,
+                    default='checkpoints/{run_name}')
+parser.add_argument('--checkpoint_interval', help='Frequency to save checkpoints', type=str, default='1ep')
+
+# Load checkpoint arguments, assumes resuming the previous training run instead of fine-tuning
+parser.add_argument('--load_checkpoint_path', help='Path to the checkpoint to load', type=str)
+
+# Recipes
+parser.add_argument('--recipe_name',
+                    help='Either "mild", "medium" or "spicy" in order of increasing training time and accuracy',
+                    type=str,
+                    choices=['mild', 'medium', 'spicy'])
+
+# Logger parameters: progress bar logging is used by default
+# Only has Weights and Biases option to reduce the number of arguments. Other loggers can be substituted in the script
+parser.add_argument('--wandb_logger', help='Whether or not to log results to Weights and Biases', action='store_true')
+parser.add_argument('--wandb_entity', help='WandB entity name', type=str)
+parser.add_argument('--wandb_project', help='WandB project name', type=str)
+parser.add_argument('--wandb_run_name', help='WandB run name', type=str)
+
+# Trainer arguments
+parser.add_argument('--run_name', help='Name of the training run used for checkpointing and other logging', type=str)
+parser.add_argument('--seed', help='Random seed', type=int, default=17)
+parser.add_argument('--max_duration',
+                    help='Duration to train specified as a Time string',
+                    type=Time.from_timestring,
+                    default='90ep')
+parser.add_argument('--eval_interval',
+                    help='How frequently to run evaluation on the validation set specified as a Time string',
+                    type=Time.from_timestring,
+                    default='1ep')
+
+# Local storage checkpointing
+args = parser.parse_args()
+
+
+def _main():
+
+    # Divide batch sizes by number of devices if running multi-gpu training
+    if dist.get_world_size():
```

Comment:
Our get_world_size() [docs](https://docs.mosaicml.com/en/v0.9.0/api_reference/composer.utils.dist.html?highlight=get_world_size#composer.utils.dist.get_world_size) aren't that informative, but the torch function that this wraps seems to return either an int >= 1 if run in part of a process group and -1 otherwise. Current logic would always evaluate to True and choke on the -1.

```suggestion
    if dist.get_world_size() > 1:
```